tools/setup_main.py.in

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Setup script for PyPI - placed in the sdist
+
+from setuptools import setup
+
+cmdclass = {}
+$extra_cmd
+
+setup(
+    name="pybind11",
+    version="$version",
+    download_url='https://github.com/pybind/pybind11/tarball/v$version',
```

Comment:
Importing from the local package in setup.py is in general a bad idea. If you were to add something that gets imported in `__init__.py` that requires a dependency, then that dependency has to be available (via PEP 518 or some other way). Even though they added a `attr: version` to setup.cfg explicitly for this purpose, they added a warning that you had to ensure that no third party dependencies got triggered. That warning is now gone, since they now parse the AST looking for a simple version assignment, and then fall back to the manual import if the AST search does not work. I think this change happened after Python 2 support was dropped, though. And, anyway, hardcoding it here is foolproof, and really easy to do with the existing infrastructure. By the way, setuptools_scm uses [templates](https://github.com/pypa/setuptools_scm/blob/master/src/setuptools_scm/__init__.py) and writes a temporary version file out too.